create/templates/shared/package.json

```diff
@@ -14,5 +14,8 @@
   "scripts": {
     "build": "shopify-cli-extensions build",
     "develop": "shopify-cli-extensions develop"
-  }
+  }{{ if and .Development.UsesTypeScript .Development.UsesReact }},
+  "resolutions": {
```

Comment:
`resolutions` is just for Yarn. I don't know what the current CLI's policy is on supporting other package managers, but I know for CLI 3.0 they intend to support the "big three" (npm, pnpm, yarn). npm and pnpm both use [`overrides` instead](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides). Is it possible for us to detect and provide only the correct field?